pkg/bosh/manifest/dns.go

```diff
@@ -0,0 +1,323 @@
+	serviceNames := dns.FindServiceNames(instanceGroupName, deploymentName)
+	if len(serviceNames) == 0 {
+		return serviceName(instanceGroupName, deploymentName, 63)
+	}
+	return serviceNames[0]
+}
+
+// ConfigurePod see interface
+func (dns *boshDomainNameService) ConfigurePod(podSpec *v1.PodSpec) {
+	if dns.localDNSIP == "" {
+		panic("BoshDomainNameService: ConfigurePod called before Reconcile")
+	}
+	podSpec.DNSPolicy = v1.DNSNone
+	podSpec.DNSConfig = &v1.PodDNSConfig{
+		Nameservers: []string{dns.localDNSIP},
+		Searches:    []string{fmt.Sprintf("%s.svc.cluster.local", dns.namespace), "service.cf.internal"},
+	}
+}
+
+// Reconcile see interface
+func (dns *boshDomainNameService) Reconcile(ctx context.Context, c client.Client, setOwner func(object v12.Object) error) error {
+
+	rewrites := ""
+	for _, alias := range dns.aliases {
+		for _, target := range alias.Targets {
+			serviceName := serviceName(target.InstanceGroup, dns.namespace, 63)
+			rewrites = rewrites + fmt.Sprintf("  rewrite name exact %s %s.%s.svc.cluster.local\n", serviceName, strings.Split(alias.Domain, ".")[0], dns.namespace)
+			rewrites = rewrites + fmt.Sprintf("  rewrite name exact %s.%s %s.%s.svc.cluster.local\n", serviceName, dns.namespace, strings.Split(alias.Domain, ".")[0], dns.namespace)
+		}
+	}
+	metadata := v12.ObjectMeta{
+		Name:      "bosh-dns",
+		Namespace: dns.namespace,
+		Labels:    map[string]string{"app": "bosh-dns"},
+	}
+
+	configMap := v1.ConfigMap{
+		ObjectMeta: metadata,
+		Data:       map[string]string{"Corefile": fmt.Sprintf(corefile, rewrites, dns.namespace, dns.rootDNSIP())},
+	}
+	service := v1.Service{
+		ObjectMeta: metadata,
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{Name: "dns", Port: 53, Protocol: "UDP", TargetPort: intstr.FromInt(8053)},
+				{Name: "dns-tcp", Port: 53, Protocol: "TCP", TargetPort: intstr.FromInt(8053)},
+				{Name: "metrics", Port: 9153, Protocol: "TCP", TargetPort: intstr.FromInt(9153)},
+			},
+			Selector: map[string]string{"app": "bosh-dns"},
+			Type:     "ClusterIP",
+		},
+	}
+
+	var mode int32 = 420
+	deployment := v13.Deployment{
+		ObjectMeta: metadata,
+		Spec: v13.DeploymentSpec{
+			Selector: &v12.LabelSelector{
+				MatchLabels: map[string]string{"app": "bosh-dns"},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: v12.ObjectMeta{
+					Labels: map[string]string{"app": "bosh-dns"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "coredns",
+							Args:  []string{"-conf", "/etc/coredns/Corefile"},
+							Image: "coredns/coredns:1.6.3",
+							Ports: []v1.ContainerPort{
+								{ContainerPort: 8053, Name: "dns-udp", Protocol: "UDP"},
+								{ContainerPort: 8053, Name: "dns-tcp", Protocol: "TCP"},
+								{ContainerPort: 9153, Name: "metrics", Protocol: "TCP"},
+							},
+							VolumeMounts: []v1.VolumeMount{
+								{MountPath: "/etc/coredns", Name: "bosh-dns-volume", ReadOnly: true},
```

Comment:
Ideally, `bosh-dns-volume`, `bosh-dns` and `Corefile` should be a consts/variables.

Reply:
OK